spec/mac_app_store_dumper_spec.rb

```diff
@@ -40,12 +40,63 @@
     before do
       Bundle::MacAppStoreDumper.reset!
       allow(Bundle).to receive(:mas_installed?).and_return(true)
-      allow(Bundle::MacAppStoreDumper).to receive(:`).and_return("123 foo\n456 bar\n789 baz")
+      allow(Bundle::MacAppStoreDumper).to receive(:`).and_return("123 foo (1.0)\n456 bar (2.0)\n789 baz (3.0)")
     end
     subject { Bundle::MacAppStoreDumper }
 
     it "returns list %w[foo bar baz]" do
       expect(subject.apps).to eql([["123", "foo"], ["456", "bar"], ["789", "baz"]])
     end
   end
+
+  context "with invalid apps details" do
+    let(:invalid_mas_output) do
+      <<~HEREDOC
+        497799835 Xcode (9.2)
+        425424353 The Unarchiver (4.0.0)
+        08981434 iMovie (10.1.8)
+        Install macOS High Sierra (13105)
+        409201541 Pages (7.1)
+        123456789 123AppNameWithNumbers (1.0)
+        409203825 Numbers (5.1)
+      HEREDOC
+    end
+
+    let(:expected_app_details_array) do
+      [
+        ["497799835", "Xcode"],
+        ["425424353", "The Unarchiver"],
+        ["08981434", "iMovie"],
+        ["409201541", "Pages"],
+        ["123456789", "123AppNameWithNumbers"],
+        ["409203825", "Numbers"],
+      ]
+    end
+
+    let(:expected_mas_dumped_output) do
+      <<~HEREDOC
+        mas "123AppNameWithNumbers", id: 123456789
+        mas "iMovie", id: 08981434
+        mas "Numbers", id: 409203825
+        mas "Pages", id: 409201541
+        mas "The Unarchiver", id: 425424353
+        mas "Xcode", id: 497799835
+      HEREDOC
+    end
+
+    before do
+      Bundle::MacAppStoreDumper.reset!
+      allow(Bundle).to receive(:mas_installed?).and_return(true)
+      allow(Bundle::MacAppStoreDumper).to receive(:`).and_return(invalid_mas_output)
+    end
+    subject { Bundle::MacAppStoreDumper }
+
+    it "returns only valid apps" do
+      expect(subject.apps).to eql(expected_app_details_array)
+    end
+
+    it "dumps excluding invalid apps" do
+      expect(subject.dump).to eq(expected_mas_dumped_output.strip)
```

Comment:
The `strip` here is an unfortunately side effect of the `HEREDOC`s and having a `\n` on the end. I tried various combinations of `%Q`/`%q`/escaping and nothing seemed to be as readable as the `HEREDOC` so I thought this was a worthy trade off.